overlord/snapshotstate/snapshotstate.go

```diff
@@ -230,30 +230,20 @@ func Restore(st *state.State, setID uint64, snapNames []string, users []string)
 	ts = state.NewTaskSet()
 
 	for _, summary := range summaries {
-		if summary.snapID == "" {
-			// snapshotted snap was unasserted, never mind
-			continue
-		}
-		snapst, ok := all[summary.snap]
-		if !ok {
-			// snap not installed
-			continue
-		}
-		sideInfo := snapst.CurrentSideInfo()
-		if sideInfo == nil {
-			// ... shouldn't happen (but also not installed)
-			continue
-		}
-		if sideInfo.SnapID == "" {
-			// current snap is unasserted, never mind then
-			continue
-		}
-		if sideInfo.SnapID != summary.snapID {
-			return nil, nil, fmt.Errorf("cannot restore snapshot over id change: %.7s… → %.7s…", summary.snapID, sideInfo.SnapID)
+		if snapst, ok := all[summary.snap]; ok {
+			info, err := snapst.CurrentInfo()
+			if err != nil {
+				// how?
+				return nil, nil, fmt.Errorf("unexpected error while reading snap info: %v", err)
+			}
+			if !info.Epoch.CanRead(&summary.epoch) {
+				return nil, nil, fmt.Errorf("cannot restore snapshot to incompatible epoch: %s → %s", &summary.epoch, &info.Epoch)
```

Comment:
I feel like wording here may be a bit confusing, though I cannot come up with anything better that isn't much longer at the same time.

Maybe something like `cannot restore snapshot from epoch %s to incompatible epoch %s` (though it still feels a bit off).

Reply:
`"cannot restore snapshot for %s: current epoch (%s) cannot read snapshot epoch (%s)"`?

Reply:
I tweaked it some more: `"cannot restore snapshot for %q: current snap (epoch %s) cannot read snapshot data (epoch %s)"`

And I changed the ID one to match: `"cannot restore snapshot for %q: current snap (ID %.7s…) does not match snapshot (ID %.7s…)"`